controllers/volumereplicationgroup_controller.go

```diff
@@ -665,15 +661,17 @@ func (v *VRGInstance) reconcileVRForDeletion(pvc *corev1.PersistentVolumeClaim,
 		return requeue
 	}
 
-	if err := v.preparePVCForVRDeletion(pvc, log); err != nil {
-		log.Info("Requeuing due to failure in preparing PersistentVolumeClaim for VolumeReplication deletion",
+	// Deleting VR first may end-up recreating the VR if reconcile for this PVC is interrupted, but that is better than
```

Comment:
Question: If the reconcile for this PVC is interrupted, why do you think it will end up recreating the VR? isn't it protected by [this line ](https://github.com/RamenDR/ramen/blob/400fddb4c913a1e9f53af02d51a71fe5fa51ce53/controllers/volumereplicationgroup_controller.go#L377). IOW, the VR should never be created if the VRG is being deleted.  So as far as I see, there is no issue here.

Reply:
Oh, I think it is [this line](https://github.com/RamenDR/ramen/blob/400fddb4c913a1e9f53af02d51a71fe5fa51ce53/controllers/volumereplicationgroup_controller.go#L661). Ignore my comment then.

Reply:
:) ack! I was just creating this comment... processForDeletion -> reconcileVRsForDeletion -> reconcileVRForDeletion -> processVRAsPrimary/Secondary -> createOrUpdateVR...